### ISSUE TITLE
grant database role 

### ISSUE THREAD

**Issue by eraliod** [Snowflake-Labs/terraform-provider-snowflake#2034]
**Is your feature request related to a problem? Please describe.**

The current resources "snowflake_grant_privileges_to_role" and "snowflake_role_grants" does not appear to allow for a statement such as:
`grant database role snowflake.usage_viewer to role test_role`
which doesn't follow the privilege grants paradigm of
grant `privileges` on `object_type` `object_name` to role `role_name`

The objective here is to grant a database role to an account role

**Describe the solution you'd like**

We would either need a new resource or adapt the grant_privileges_to_role to behave differently for the "DATABASE ROLE" type privilege

**Describe alternatives you've considered**

I am currently using snowsql_exec as an alternative
```
resource "snowsql_exec" "snowflake_database_grants" {
  name = random_pet.snowsql_query_identifier.id
  create {
    statements = "GRANT DATABASE ROLE SNOWFLAKE.USAGE_VIEWER TO ROLE TEST_ROLE"
  }
  delete {
    statements = "REVOKE DATABASE ROLE SNOWFLAKE.USAGE_VIEWER FROM ROLE TEST_ROLE"
  }
  depends_on = [
    random_pet.snowsql_query_identifier,
   ]
}
```

**Additional context**

**Comment by abid1995** [comment]
I think database grant was released with v0.70.0 with https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2014, but there are no documentation yet on how we can use it. Also another related issue https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2018 is available.
Currently, also using `snowsql` as a workaround similar like @eraliod

**Comment by eraliod** [comment]
Thanks @abid1995 but I think the snowflake_database_role resource released in v0.70.0 is to create a database role, not to grant a database role to another role. I did +1 the issue you mentioned, as it does appear the documentation is missing for that resource.

**Comment by abid1995** [comment]
> #2014

Thanks @eraliod , 🤔 saw the PR https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2014 released in v0.70.0 which references to the `grant database role`

**Comment by abid1995** [comment]
> Thanks @abid1995 but I think the snowflake_database_role resource released in v0.70.0 is to create a database role, not to grant a database role to another role. I did +1 the issue you mentioned, as it does appear the documentation is missing for that resource.

Thanks @eraliod , 🤔 saw the PR https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/2014 released in v0.70.0 which references to the `grant database role`

**Comment by eraliod** [comment]
on second review, this is a duplicate from #2014 as mentioned in the comments above
